tests/phpunit/tests/l10n/wpWordCount.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+/**
+ * @group l10n
+ * @group i18n
+ *
+ * @covers ::wp_word_count
+ */
+class Tests_L10n_wpWordcount extends WP_UnitTestCase {
+	protected static $settings;
+
+	public static function wpSetUpBeforeClass() {
+		self::$settings = array(
+			'shortcodes' => array( 'shortcode' ),
+		);
+	}
+
+	/**
+	 * @ticket 57987
+	 *
+	 * @dataProvider data_get_string_variations
+	 *
+	 * @param string $text     Text to count elements in.
+	 * @param int    $expected Expected word count.
+	 */
+	public function test_wp_word_count_should_return_the_number_of_words( $text, $expected ) {
+		$this->assertSame( wp_word_count( $text, 'words', self::$settings ), $expected['words'] );
```

Comment:
These arguments are in the wrong order.

`assertSame( $expected, $actual  );` is the correct order.

Reply:
Fixed by 310ed7a